### PR TITLE
Change default domain of SR.var()

### LINKED ISSUE #36833: factor ignores complex conjugate
### Steps To Reproduce

1. I run SageMath version 10.2, Release Date: 2023-12-03.
3. When I enter the following commands:
```
y = SR.var("y") # or y = var("y")
f = y * conjugate(y)
f.factor()
```

4. This returns y^2 which is clearly not true for y being complex.

### Expected Behavior

The expected result for this would be to just return y * conjugate(y) as by documentation when declaring y as a variable of SR it is assumed that y is in the complex plain.

### Actual Behavior

The factor function treats y as a real variable and thus returns the wrong result of y^2.

### Additional Information

_No response_

### Environment

```markdown
- **OS**: Arch Linux
- **Kernel**: Linux 6.6.4-arch1-1
- **Sage Version**: 10.2
- **Python Version**: 3.11.6
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by RuchitJagodara**
This patch fixes #36833 by changing the default domain of SR.var()
function to complex. By this way, every variable whose domain is not defined will be considered
in complex plane, by default.

The issue was with the default value only because whenever we were defining a variable without specifying any domain it was calling symbol() function with domain=None; Subsequently, it calls new_Expression_symbol() function with domain=None. In this code, I found that it was not giving any specific domain to the variable and so when we define a new variable using SR.var() function, it is passing the is_real() (Because it does not have any specific domain) test but giving the wrong answer on factor() function, if we assume that the variable is in complex plane by default which is correct according to the documentation.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.

### CODE REVIEW

File: src/sage/symbolic/ring.pyx [hunk 1]
```diff
@@ -862,6 +862,16 @@ cdef class SymbolicRing(sage.rings.abc.SymbolicRing):
             Traceback (most recent call last):
             ...
             ValueError: cannot specify n for multiple symbol names
+        
+        Check that :trac:`36833` is fixed: Variables whose domain is not
+        defined should be considered in complex plane as described in documentation::
```
Comment (dimpase) [inline src/sage/symbolic/ring.pyx]: Could you point out where in documentation you see this " considered in complex plane" ?

**Comment by dimpase** [comment]
we might see lots of doctest errors now, not sure.

File: src/sage/symbolic/ring.pyx (hunk 1 above)
Comment (RuchitJagodara) [reply src/sage/symbolic/ring.pyx]: Screenshot from 2023-12-10 19-07-00

**Comment by RuchitJagodara** [comment]
Then, I think it would be good if we change the factor function instead of doing this. I'll change it and will update.

**Comment by RuchitJagodara** [comment]
@dimpase Do you know any function that can return the domain of a symbolic variable?

**Comment by dimpase** [comment]
> @dimpase Do you know any function that can return the domain of a symbolic variable?

no, I don't.

By the way, I just saw the place where it's claimed that `var` by default returns
complex variable - it's in the output `sage: var?` - so you can find it in the code by
e.g. looking at the bottom of `sage: var??`

**Comment by RuchitJagodara** [comment]
Can't we implement a new class for variables of symbolic ring? By this method, we will be able to track domain of the variable and once we are able to access the domain of variable, I think it will be easier to handle these type of issues.

File: src/sage/symbolic/ring.pyx (hunk 1 above)
Comment (dimpase) [reply src/sage/symbolic/ring.pyx]: thanks!

**Comment by dimpase** [comment]
I don't think it's a good idea to make `domain="complex"` the default. It seems that most everything in Sage assumes it to be real. E.g. if you make this change then differentiating with default variables will become complex differentiation, which is a rather different story, mathematically - and Sage does not do it.
E.g.
```python
sage: y = SR.var("y",domain="complex")
sage: f(y) = y * conjugate(y)
sage: f.diff(y)
y |--> y*diff(conjugate(y), y) + conjugate(y)
```
doesn't make much sense, as in fact $\frac{d\overline{y}}{dy}$ does not exist (e.g. as Cauchy equations don't hold).

**Comment by RuchitJagodara** [comment]
Hmm, got it. But when we declare a variable without declaring any domain, it is clear that it is not real from the  following,

```python
sage: n = SR.var("n"); n
n
sage: n.is_real()
False
``` 

So it simply implies that it is complex. So I am not getting the point that what is wrong here, if the variable is not real then it should not be simplified to y^2, anyhow. But factor function is doing so. I think the error is occurring with the maxima. Actually maxima is simplifying many functions which should not be simplified like 0^n is being converted to 0 (this is referenced from the issue #36838 ) , here y * conjugate(y) is being converted to y^2.

**Comment by dimpase** [comment]
On 12 December 2023 15:48:25 GMT, Ruchit Jagodara ***@***.***> wrote:
>Hmm, got it. But when we declare a variable without declaring any domain, it is clear that it is not real from the  following,
>
>```python
>sage: n = SR.var("n"); n
>n
>sage: n.is_real()
>False
>``` 
>
>So it simply implies that it is complex. 

it'd imply this if our docs were correct:-)

By default  `domain=None`, and it could be that `is_real()` just checked that.



> So I am not getting the point that what is wrong here, if the variable is not real then it should not be simplified to y^2, anyhow. But factor function is doing so. I think the error is occurring with the maxima. Actually maxima is simplifying many functions which should not be simplified like 0^n is being converted to 0 (this is referenced from the issue #36838 ) , here y * conjugate(y) is being converted to y^2. 

maxima has no concept of a domain for a variable, it seems to me.

**Comment by RuchitJagodara** [comment]
> maxima has no concept of a domain for a variable, it seems to me.

But it has to be, because when we do operation like below 

```python
sage: n = var('n',domain="complex"); n
n
sage: f1 = n * conjugate(n); f1
n*conjugate(n)
sage: f1.factor()
n*conjugate(n)
```

Here, we can clearly see that it is giving correct answer if we declare domain="complex", but it is not giving the same answer when we do it as below, which means it has something to do with the domain.

```python
sage: b = var("b")
sage: f2 = b * conjugate(b); f2
b*conjugate(b)
sage: f2.factor()
b^2
```

**Comment by dimpase** [comment]
> > maxima has no concept of a domain for a variable, it seems to me.
> 
> But it has to be, because when we do operation like below
> 
> ```python
> sage: n = var('n',domain="complex"); n
> n
> sage: f1 = n * conjugate(n); f1
> n*conjugate(n)
> sage: f1.factor()
> n*conjugate(n)
> ```
> 
> Here, we can clearly see that it is giving correct answer if we declare domain="complex", but it is not giving the same answer when we do it as below, which means it has something to do with the domain.
> 
> ```python
> sage: b = var("b")
> sage: f2 = b * conjugate(b); f2
> b*conjugate(b)
> sage: f2.factor()
> b^2
> ```

Sorry, indeed, you are right. Maxima has [domain](https://maxima.sourceforge.io/docs/manual/maxima_46.html#index-domain), which is a domain for computation to be done (although it's not to declare that a variable is real or complex) , and  a similar thing called `assume` (which allows things like `assume(x>0)`, or indeed `x` to be real).

Better, it has [declare](https://maxima.sourceforge.io/docs/manual/maxima_49.html#index-conjugate) - which seems to be exactly what is needed here. So one can do
```
$ maxima 
Maxima 5.47.0 https://maxima.sourceforge.io
using Lisp SBCL 2.3.9
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i4) declare(x, complex);
(%o4)                                done
(%i5) x*conjugate(x);
(%o5)                           x conjugate(x)
(%i6) declare(y, real);
(%o6)                                done
(%i7) y*conjugate(y);
                                       2
(%o7)                                 y
(%i8) 

```

**Comment by RuchitJagodara** [comment]
Hmm... But in sage when we are doing this 
```python
sage: y = SR.var("y","real"); y
y
sage: f = y * conjugate(y); f
y*conjugate(y)
sage: f
y*conjugate(y)
sage: f.simplify_full()
y^2
``` 

Here, you can see that for variable whose domain is real, it is not simplifying conjuage(y) and for that I have to use simplify_full() function but I think this should be done when we were multiplying y with conjuage(y) so the value of f should be y^2 not y*conjugate(y).

**Comment by RuchitJagodara** [comment]
I found an issue where one had similar problem with maxima, [discussion link](https://github.com/sagemath/sage/issues/36410#issuecomment-1763476272)

**Comment by RuchitJagodara** [comment]
Should I change the documentation instead, which may specify that the default domain of `var` is None? This means that the variable will not have any specific domain. Or should I just leave it as it is?

**Comment by dimpase** [comment]
Yes, please, we are not going to make the default domain `complex`. Please fix the documentation instead, to specify that the default domain is `None`.

**Comment by RuchitJagodara** [comment]
Sorry for the delay! Actually, I was checking if there is any other fix that can be applied, but I think we cannot do anything as the problem is with maxima only. So, I have changed the documentation.

Can you please review my changes now ?

**Review by dimpase — APPROVED** [review]
OK

**Pull request merged**